### DISCUSSION TITLE
Android Dependencies

### DISCUSSION THREAD

**Discussion by ashrobo** [MobileNativeFoundation/discussions#173] [General]
Hi

Really interested to know how larger companies manage updating to newer versions of Android Studio including the build tooling, gradle, gradle plug-in updates and the various toolchain  dependencies. 

As you know, Android Studio downloads most of its dependencies automatically however we have a strict policy of not allowing developers to download code or applications externally. Do any other companies have similar restrictions?

We started hosting dependencies internally on our own maven feed however collating the list of required dependencies and version becomes a much trickier task each year as Android Studio finds new and novel ways to cache all of the artefacts in various different locations across the system, as well as new ways of constructing the dependency tree. Breaking changes to our gradle scripts with each update are also a big pain point.

Thanks in advance.

**Comment by waliahimanshu** [comment]
>As you know, Android Studio downloads most of its dependencies automatically however we have a strict policy of not allowing developers to download code or applications externally.

Which dependencies do you mean? IDE, plugin updates, SDK tools update or third-party dependencies declared in gradle ?

**Reply by ashrobo** [reply]
Mainly the ones that end up in .gradle/caches/files-2.1. I think this includes both third party dependencies and Android studio/build tool dependencies. Essentially everything in that folder cannot just be downloaded automatically from the internet through Android Studio it has to be specifically requested using URLs to central Maven etc and scanned in advance. 

Is it common for other companies to allow tools to download files/code from the internet onto internal systems? If not I’m interested in how this can be managed in a less manual way.

**Reply by waliahimanshu** [reply]
I have not seen this restriction in other companies, certainly, this is not common.
What is the reason you can not download the dependencies from the central [google ](https://maven.google.com/web/index.html)and maven repositories?

If its security, you could scan the Gradle dependencies for OWASP vulnerabilities for early detection
https://plugins.gradle.org/plugin/org.owasp.dependencycheck
